smdebug/profiler/profiler_config_parser.py

```diff
@@ -356,3 +427,5 @@ def reset_profiler_config_parser():
     """
     global _profiler_config_parser
     _profiler_config_parser = None
+
+    reset_python_profiler()
```

Comment:
if this function is only intended to be called by the test framework, can we indicate that with the fn name?
Maybe with `_` as a prefix?

Reply:
We should only prefix with `_` for functions called in the same module. 

But perhaps we can move this reset function to the test utils (along with the Python profiling one)

Reply:
We should only prefix with `_` if the function is only used inside the same module. This function gets called outside the module.

Reply:
Added to `reset_python_profiler`'s docstring to make it clear that it is only intended to be used for testing.